transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java

```diff
@@ -656,6 +657,20 @@ public V apply(final K k) {
     private final Map<Store<?, ?>, SoftLockValueCombinedSerializerLifecycleHelper> createdStores = new ConcurrentWeakIdentityHashMap<Store<?, ?>, SoftLockValueCombinedSerializerLifecycleHelper>();
 
     @Override
+    public int rank(final Set<ResourceType> resourceTypes, final Collection<ServiceConfiguration<?>> serviceConfigs) {
```

Comment:
This seems wrong... XAStore should be doing a recursive call on StoreSupport.selectStoreProvider that is passed a StoreProvider that filters out this provider from the set of returned store providers.